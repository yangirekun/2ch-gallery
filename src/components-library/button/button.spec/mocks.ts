export const mockProps = [
  {
    id: "button-test-id",
    label: "button-test-text",
    className: "button-test-class",
    disabled: false,
    onClick: jest.fn(),
  },
  {
    id: "button-test-id",
    label: "button-test-text",
    className: "button-test-class",
    disabled: true,
    onClick: jest.fn(),
  },
  {
    id: "button-test-id",
  },
];
