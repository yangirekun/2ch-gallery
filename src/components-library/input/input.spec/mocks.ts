export const mockProps = [
  {
    id: "input-test-id",
    label: "input-test-label",
    className: "input-test-class",
    placeholder: "input-test-placeholder",
    value: "input-test-value",
    onChange: jest.fn(),
  },
];
