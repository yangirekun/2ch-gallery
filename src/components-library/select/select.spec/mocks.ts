const mockList = [
  { id: "b", label: "Бред" },
  { id: "vg", label: "Video Games General" },
  { id: "po", label: "Политика" },
  { id: "fag", label: "Фагготрия" },
  { id: "news", label: "Новости" },
  { id: "sex", label: "Секс и отношения" },
  { id: "ma", label: "Манга" },
  { id: "hw", label: "Компьютерное железо" },
  { id: "biz", label: "Бизнес" },
  { id: "v", label: "Video Games" },
  { id: "wrk", label: "Работа и карьера" },
  { id: "ftb", label: "Футбол" },
  { id: "pr", label: "Программирование" },
  { id: "cc", label: "Криптовалюта" },
  { id: "rf", label: "Убежище" },
  { id: "au", label: "Автомобили" },
  { id: "alco", label: "Алкоголь" },
  { id: "2d", label: "Аниме/Беседка" },
  { id: "mov", label: "Фильмы" },
  { id: "ja", label: "Японская культура" },
  { id: "soc", label: "Общение" },
  { id: "gg", label: "Хорошие девушки" },
  { id: "mobi", label: "Мобильные устройства и приложения" },
  { id: "me", label: "Медицина" },
  { id: "sp", label: "Спорт" },
  { id: "cg", label: "Консоли" },
  { id: "bo", label: "Книги" },
  { id: "gsg", label: "Grand Strategy Games" },
  { id: "a", label: "Аниме" },
  { id: "bi", label: "Велосипеды" },
  { id: "wm", label: "Военная техника" },
  { id: "fa", label: "Мода и стиль" },
  { id: "fiz", label: "Физкультура" },
  { id: "by", label: "Беларусь" },
  { id: "mo", label: "Мотоциклы" },
  { id: "hi", label: "История" },
  { id: "psy", label: "Психология и психиатрия" },
  { id: "w", label: "Оружие" },
  { id: "brg", label: "Браузерные игры" },
  { id: "tes", label: "The Elder Scrolls" },
  { id: "bg", label: "Настольные игры" },
  { id: "hry", label: "Highlights of Russian-speaking youtube" },
  { id: "gd", label: "Gamedev" },
  { id: "s", label: "Программы" },
  { id: "pa", label: "Живопись" },
  { id: "fd", label: "Фэндом" },
  { id: "d", label: "Скла/d/ грязи" },
  { id: "fs", label: "Фэнтези" },
  { id: "fet", label: "Фетиш" },
  { id: "em", label: "Другие страны" },
  { id: "dr", label: "Дневнички" },
  { id: "re", label: "Религия" },
  { id: "spc", label: "Космос и астрономия" },
  { id: "diy", label: "Хобби" },
  { id: "mus", label: "Музыканты" },
  { id: "socionics", label: "Соционика и психософия" },
  { id: "c", label: "Комиксы и мультфильмы" },
  { id: "un", label: "Образование" },
  { id: "p", label: "Фотография" },
  { id: "fg", label: "Трапы" },
  { id: "e", label: "Extreme Pron" },
  { id: "di", label: "Столовая" },
  { id: "td", label: "Трёхмерная графика" },
  { id: "kpop", label: "K-pop" },
  { id: "obr", label: "Offline Battle Rap" },
  { id: "tv", label: "Сериалы" },
  { id: "hh", label: "Hip-Hop" },
  { id: "ra", label: "Радиотехника" },
  { id: "sn", label: "Паранормальные явления" },
  { id: "wh", label: "Warhammer" },
  { id: "dom", label: "Домострой" },
  { id: "fur", label: "Фурри" },
  { id: "ga", label: "Геи" },
  { id: "mu", label: "Музыка" },
  { id: "h", label: "Хентай" },
  { id: "hc", label: "Hardcore" },
  { id: "hv", label: "Халява в интернете" },
  { id: "sci", label: "Наука" },
  { id: "ho", label: "Прочий хентай" },
  { id: "zog", label: "Теории заговора" },
  { id: "fl", label: "Иностранные языки" },
  { id: "mmo", label: "Massive multiplayer online games" },
  { id: "mg", label: "Магия" },
  { id: "t", label: "Техника" },
  { id: "izd", label: "Графомания" },
  { id: "vn", label: "Визуальные новеллы" },
  { id: "law", label: "Право и правоприменители" },
  { id: "qtr4", label: "qtr4" },
  { id: "asmr", label: "ASMR" },
  { id: "vape", label: "Электронные сигареты" },
  { id: "ruvn", label: "Российские визуальные новеллы" },
  { id: "ph", label: "Философия" },
  { id: "wr", label: "Текстовые авторские рпг" },
  { id: "rm", label: "Rozen Maiden" },
  { id: "ukr", label: "/Ukr/aine — Україна" },
  { id: "wow", label: "World of Warcraft" },
  { id: "es", label: "Бесконечное Лето" },
  { id: "whn", label: "Warhammer Nihilus" },
  { id: "ne", label: "Животные и природа" },
  { id: "de", label: "Дизайн" },
  { id: "aa", label: "Аниме арт" },
  { id: "web", label: "Web-мастера" },
  { id: "out", label: "Активный отдых" },
  { id: "media", label: "Анимация" },
  { id: "m", label: "Меха" },
  { id: "se", label: "Самообразование" },
  { id: "kz", label: "Казачан" },
  { id: "o", label: "Рисовач" },
  { id: "ch", label: "Чатики и конфочки" },
  { id: "math", label: "Математика" },
  { id: "ew", label: "Конец света" },
  { id: "mlp", label: "My Little Pony" },
  { id: "pvc", label: "Фигурки" },
  { id: "sf", label: "Научная фантастика" },
  { id: "tr", label: "Транспорт и авиация" },
  { id: "wp", label: "Обои и высокое разрешение" },
  { id: "mc", label: "Minecraft" },
  { id: "to", label: "Touhou" },
  { id: "r", label: "Просьбы" },
  { id: "br", label: "Барахолка" },
  { id: "trv", label: "Путешествия и отдых" },
  { id: "gb", label: "Азартные игры" },
  { id: "moba", label: "Multiplayer Online Battle Arena games" },
  { id: "abu", label: "abu" },
  { id: "cul", label: "Культура" },
  { id: "old", label: "Олдфаги" },
  { id: "guro", label: "Шок-контент" },
  { id: "ussr", label: "СССР" },
  { id: "jsf", label: "Japanese Street Fashion" },
  { id: "sw", label: "Star Wars" },
  { id: "ya", label: "Яой" },
  { id: "r34", label: "Правило 34" },
  { id: "gabe", label: "Gabe Logan Newell" },
  { id: "cute", label: "Милое" },
  { id: "8", label: "8bit" },
  { id: "mlpr", label: "My Little Pony Roleplay" },
  { id: "ro", label: "Рогалики" },
  { id: "who", label: "Doctor Who" },
  { id: "srv", label: "Серверы" },
  { id: "electrach", label: "Электронная музыка" },
  { id: "ing", label: "Инженерное дело" },
  { id: "got", label: "Песнь льда и пламени" },
  { id: "crypt", label: "Криптоанархия" },
  { id: "lap", label: "Лапник (4x4 клуб)" },
  { id: "smo", label: "Курение" },
  { id: "hg", label: "Хентайные игры/hentai games" },
  { id: "sad", label: "Садоводство" },
  { id: "fi", label: "Рыбалка" },
  { id: "nvr", label: "Новороссия: ЛНР, ДНР" },
  { id: "ind", label: "Инди" },
  { id: "ld", label: "Осознанные сновидения" },
  { id: "fem", label: "Феминизм" },
  { id: "vr", label: "Виртуальная реальность" },
  { id: "arg", label: "ARG/игры в альтернативной реальности" },
  { id: "char", label: "Сетевые персонажи" },
  { id: "pok", label: "Pokemon" },
  { id: "wwe", label: "WorldWide Wrestling Universe" },
  { id: "int", label: "International" },
  { id: "catalog", label: "Статьи" },
  { id: "api", label: "API" },
];

export const mockProps = [
  {
    id: "select-test-id",
    label: "select-test-label",
    className: "select-test-class",
    placeholder: "select-test-placeholder",
    list: mockList,
    value: mockList[0].id,
    onChange: jest.fn(),
  },
];
