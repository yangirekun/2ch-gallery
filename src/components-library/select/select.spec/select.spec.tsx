import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { Select } from "../index";
import { mockProps } from "./mocks";

describe("testing <Select />", () => {
  test("it should be rendered correctly", () => {
    const renderResult = render(<Select {...mockProps[0]} />);
    const { getByTestId } = renderResult;

    expect(renderResult.asFragment()).toMatchSnapshot();
    fireEvent.click(getByTestId("select-test-id-switcher"));
    expect(renderResult.asFragment()).toMatchSnapshot();
  });

  test("it should toggle data-list", () => {
    const { container, getByTestId } = render(<Select {...mockProps[0]} />);
    const toggle = getByTestId("select-test-id-switcher");
    fireEvent.click(toggle);
    expect(
      container.querySelector(`[data-testid="select-test-id-data-list"`)
    ).toBeVisible();
    fireEvent.click(toggle);
    expect(
      container.querySelector(`[data-testid="select-test-id-data-list`)
    ).toBeNull();
  });

  test("it should filter data-list correctly", () => {
    const { getByTestId } = render(<Select {...mockProps[0]} />);
    const toggle = getByTestId("select-test-id-switcher");
    const input: any = getByTestId("select-test-id-input");
    fireEvent.click(toggle);
    const dataList = getByTestId("select-test-id-data-list");
    fireEvent.change(input, { target: { value: "Бр" } });
    expect(input.value).toBe("Бр");
    expect(dataList.children.length).toBe(4);
  });

  test("it should call onChange when value is selected", () => {
    const { getByTestId } = render(<Select {...mockProps[0]} />);
    const toggle = getByTestId("select-test-id-switcher");
    const input: any = getByTestId("select-test-id-input");
    fireEvent.click(toggle);
    fireEvent.change(input, { target: { value: "Бред" } });
    const listItem = getByTestId("select-test-id-data-list-item-0");
    fireEvent.click(listItem);
    expect(mockProps[0].onChange).toBeCalledTimes(1);
  });
});
