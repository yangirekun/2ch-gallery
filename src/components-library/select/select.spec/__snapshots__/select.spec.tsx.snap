// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <Select /> it should be rendered correctly 1`] = `
<DocumentFragment>
  <div
    class="select select-test-class"
    data-testid="select-test-id-wrapper"
  >
    <label
      class="select__label"
      data-testid="select-test-id-label"
      for="select-test-id"
    >
      select-test-label
    </label>
    <input
      class="select__field"
      data-testid="select-test-id-input"
      id="select-test-id"
      placeholder="select-test-placeholder"
      type="text"
      value="Бред"
    />
    <img
      alt="dropdown-arrow"
      class="select__switcher select__switcher--collapsed"
      data-testid="select-test-id-switcher"
      src="icons8-expand-arrow-24.png"
    />
  </div>
</DocumentFragment>
`;

exports[`testing <Select /> it should be rendered correctly 2`] = `
<DocumentFragment>
  <div
    class="select select-test-class"
    data-testid="select-test-id-wrapper"
  >
    <label
      class="select__label"
      data-testid="select-test-id-label"
      for="select-test-id"
    >
      select-test-label
    </label>
    <input
      class="select__field"
      data-testid="select-test-id-input"
      id="select-test-id"
      placeholder="select-test-placeholder"
      type="text"
      value="Бред"
    />
    <img
      alt="dropdown-arrow"
      class="select__switcher select__switcher--expanded"
      data-testid="select-test-id-switcher"
      src="icons8-expand-arrow-24.png"
    />
    <ul
      class="select__data-list"
      data-testid="select-test-id-data-list"
    >
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-0"
        id="b"
      >
        Бред
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-1"
        id="vg"
      >
        Video Games General
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-2"
        id="po"
      >
        Политика
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-3"
        id="fag"
      >
        Фагготрия
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-4"
        id="news"
      >
        Новости
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-5"
        id="sex"
      >
        Секс и отношения
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-6"
        id="ma"
      >
        Манга
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-7"
        id="hw"
      >
        Компьютерное железо
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-8"
        id="biz"
      >
        Бизнес
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-9"
        id="v"
      >
        Video Games
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-10"
        id="wrk"
      >
        Работа и карьера
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-11"
        id="ftb"
      >
        Футбол
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-12"
        id="pr"
      >
        Программирование
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-13"
        id="cc"
      >
        Криптовалюта
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-14"
        id="rf"
      >
        Убежище
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-15"
        id="au"
      >
        Автомобили
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-16"
        id="alco"
      >
        Алкоголь
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-17"
        id="2d"
      >
        Аниме/Беседка
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-18"
        id="mov"
      >
        Фильмы
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-19"
        id="ja"
      >
        Японская культура
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-20"
        id="soc"
      >
        Общение
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-21"
        id="gg"
      >
        Хорошие девушки
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-22"
        id="mobi"
      >
        Мобильные устройства и приложения
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-23"
        id="me"
      >
        Медицина
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-24"
        id="sp"
      >
        Спорт
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-25"
        id="cg"
      >
        Консоли
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-26"
        id="bo"
      >
        Книги
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-27"
        id="gsg"
      >
        Grand Strategy Games
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-28"
        id="a"
      >
        Аниме
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-29"
        id="bi"
      >
        Велосипеды
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-30"
        id="wm"
      >
        Военная техника
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-31"
        id="fa"
      >
        Мода и стиль
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-32"
        id="fiz"
      >
        Физкультура
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-33"
        id="by"
      >
        Беларусь
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-34"
        id="mo"
      >
        Мотоциклы
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-35"
        id="hi"
      >
        История
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-36"
        id="psy"
      >
        Психология и психиатрия
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-37"
        id="w"
      >
        Оружие
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-38"
        id="brg"
      >
        Браузерные игры
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-39"
        id="tes"
      >
        The Elder Scrolls
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-40"
        id="bg"
      >
        Настольные игры
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-41"
        id="hry"
      >
        Highlights of Russian-speaking youtube
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-42"
        id="gd"
      >
        Gamedev
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-43"
        id="s"
      >
        Программы
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-44"
        id="pa"
      >
        Живопись
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-45"
        id="fd"
      >
        Фэндом
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-46"
        id="d"
      >
        Скла/d/ грязи
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-47"
        id="fs"
      >
        Фэнтези
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-48"
        id="fet"
      >
        Фетиш
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-49"
        id="em"
      >
        Другие страны
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-50"
        id="dr"
      >
        Дневнички
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-51"
        id="re"
      >
        Религия
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-52"
        id="spc"
      >
        Космос и астрономия
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-53"
        id="diy"
      >
        Хобби
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-54"
        id="mus"
      >
        Музыканты
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-55"
        id="socionics"
      >
        Соционика и психософия
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-56"
        id="c"
      >
        Комиксы и мультфильмы
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-57"
        id="un"
      >
        Образование
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-58"
        id="p"
      >
        Фотография
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-59"
        id="fg"
      >
        Трапы
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-60"
        id="e"
      >
        Extreme Pron
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-61"
        id="di"
      >
        Столовая
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-62"
        id="td"
      >
        Трёхмерная графика
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-63"
        id="kpop"
      >
        K-pop
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-64"
        id="obr"
      >
        Offline Battle Rap
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-65"
        id="tv"
      >
        Сериалы
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-66"
        id="hh"
      >
        Hip-Hop
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-67"
        id="ra"
      >
        Радиотехника
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-68"
        id="sn"
      >
        Паранормальные явления
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-69"
        id="wh"
      >
        Warhammer
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-70"
        id="dom"
      >
        Домострой
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-71"
        id="fur"
      >
        Фурри
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-72"
        id="ga"
      >
        Геи
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-73"
        id="mu"
      >
        Музыка
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-74"
        id="h"
      >
        Хентай
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-75"
        id="hc"
      >
        Hardcore
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-76"
        id="hv"
      >
        Халява в интернете
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-77"
        id="sci"
      >
        Наука
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-78"
        id="ho"
      >
        Прочий хентай
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-79"
        id="zog"
      >
        Теории заговора
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-80"
        id="fl"
      >
        Иностранные языки
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-81"
        id="mmo"
      >
        Massive multiplayer online games
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-82"
        id="mg"
      >
        Магия
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-83"
        id="t"
      >
        Техника
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-84"
        id="izd"
      >
        Графомания
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-85"
        id="vn"
      >
        Визуальные новеллы
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-86"
        id="law"
      >
        Право и правоприменители
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-87"
        id="qtr4"
      >
        qtr4
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-88"
        id="asmr"
      >
        ASMR
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-89"
        id="vape"
      >
        Электронные сигареты
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-90"
        id="ruvn"
      >
        Российские визуальные новеллы
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-91"
        id="ph"
      >
        Философия
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-92"
        id="wr"
      >
        Текстовые авторские рпг
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-93"
        id="rm"
      >
        Rozen Maiden
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-94"
        id="ukr"
      >
        /Ukr/aine — Україна
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-95"
        id="wow"
      >
        World of Warcraft
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-96"
        id="es"
      >
        Бесконечное Лето
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-97"
        id="whn"
      >
        Warhammer Nihilus
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-98"
        id="ne"
      >
        Животные и природа
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-99"
        id="de"
      >
        Дизайн
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-100"
        id="aa"
      >
        Аниме арт
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-101"
        id="web"
      >
        Web-мастера
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-102"
        id="out"
      >
        Активный отдых
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-103"
        id="media"
      >
        Анимация
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-104"
        id="m"
      >
        Меха
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-105"
        id="se"
      >
        Самообразование
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-106"
        id="kz"
      >
        Казачан
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-107"
        id="o"
      >
        Рисовач
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-108"
        id="ch"
      >
        Чатики и конфочки
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-109"
        id="math"
      >
        Математика
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-110"
        id="ew"
      >
        Конец света
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-111"
        id="mlp"
      >
        My Little Pony
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-112"
        id="pvc"
      >
        Фигурки
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-113"
        id="sf"
      >
        Научная фантастика
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-114"
        id="tr"
      >
        Транспорт и авиация
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-115"
        id="wp"
      >
        Обои и высокое разрешение
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-116"
        id="mc"
      >
        Minecraft
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-117"
        id="to"
      >
        Touhou
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-118"
        id="r"
      >
        Просьбы
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-119"
        id="br"
      >
        Барахолка
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-120"
        id="trv"
      >
        Путешествия и отдых
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-121"
        id="gb"
      >
        Азартные игры
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-122"
        id="moba"
      >
        Multiplayer Online Battle Arena games
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-123"
        id="abu"
      >
        abu
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-124"
        id="cul"
      >
        Культура
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-125"
        id="old"
      >
        Олдфаги
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-126"
        id="guro"
      >
        Шок-контент
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-127"
        id="ussr"
      >
        СССР
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-128"
        id="jsf"
      >
        Japanese Street Fashion
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-129"
        id="sw"
      >
        Star Wars
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-130"
        id="ya"
      >
        Яой
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-131"
        id="r34"
      >
        Правило 34
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-132"
        id="gabe"
      >
        Gabe Logan Newell
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-133"
        id="cute"
      >
        Милое
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-134"
        id="8"
      >
        8bit
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-135"
        id="mlpr"
      >
        My Little Pony Roleplay
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-136"
        id="ro"
      >
        Рогалики
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-137"
        id="who"
      >
        Doctor Who
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-138"
        id="srv"
      >
        Серверы
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-139"
        id="electrach"
      >
        Электронная музыка
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-140"
        id="ing"
      >
        Инженерное дело
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-141"
        id="got"
      >
        Песнь льда и пламени
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-142"
        id="crypt"
      >
        Криптоанархия
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-143"
        id="lap"
      >
        Лапник (4x4 клуб)
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-144"
        id="smo"
      >
        Курение
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-145"
        id="hg"
      >
        Хентайные игры/hentai games
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-146"
        id="sad"
      >
        Садоводство
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-147"
        id="fi"
      >
        Рыбалка
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-148"
        id="nvr"
      >
        Новороссия: ЛНР, ДНР
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-149"
        id="ind"
      >
        Инди
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-150"
        id="ld"
      >
        Осознанные сновидения
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-151"
        id="fem"
      >
        Феминизм
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-152"
        id="vr"
      >
        Виртуальная реальность
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-153"
        id="arg"
      >
        ARG/игры в альтернативной реальности
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-154"
        id="char"
      >
        Сетевые персонажи
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-155"
        id="pok"
      >
        Pokemon
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-156"
        id="wwe"
      >
        WorldWide Wrestling Universe
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-157"
        id="int"
      >
        International
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-158"
        id="catalog"
      >
        Статьи
      </li>
      <li
        class="select__list-item"
        data-testid="select-test-id-data-list-item-159"
        id="api"
      >
        API
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
